src/Searchbar.vue

```diff
@@ -15,38 +15,95 @@ export default {
   },
   computed: {
     primitiveData() {
+      function allMatch(info, regexes) {
+        return regexes.every((regex) => regex.test(info));
+      }
+
+      function someMatch(info, regexes) {
+        return regexes.some((regex) => regex.test(info));
+      }
+
       if (!this.data) {
         return undefined;
       }
-      const matches = [];
-      const regex = new RegExp(this.value, 'i');
+      let matches = [];
+      const regexes = this.value.split(' ').map(keyword => new RegExp(keyword, 'i'));
       this.data.forEach((entry) => {
         const { headings, src, title } = entry;
         const keywords = entry.keywords || '';
         let hasMatchingHeading = false;
+        const headingEntries = [];
+
         Object.entries(headings).forEach(([id, text]) => {
-          if (regex.test(text)) {
-            hasMatchingHeading = true;
-            matches.push({
-              heading: { id, text },
-              keywords,
-              src,
-              title,
-            });
-          }
-        });
-        if (!hasMatchingHeading) {
-          if (regex.test(title) || regex.test(keywords)) {
-            matches.push(entry);
+            const info = [title].concat(keywords).concat(text).join(' ');
+            if (allMatch(info, regexes)) {
+              if(someMatch(text, regexes)) {
+                hasMatchingHeading = true;
+              }
+              headingEntries.push({
+                heading: { id, text },
+                keywords,
+                src,
+                title,
+              });
+            }
+          });
+          if (!hasMatchingHeading) {
+            const info = [title].concat(keywords).join(' ');
+            if(allMatch(info, regexes)) {
+              matches.push(entry);
+            }
           }
-        }
+          matches = matches.concat(headingEntries);
       });
       return matches;
     },
   },
   filters: {
     highlight(value, phrase) {
-      return value.replace(new RegExp(`(${phrase})`, 'gi'), '<mark>$1</mark>');
+      function mergeIntervals(intervals) {
+        const stack = [];
+        let top = null;
```

Comment:
You can declare this below at line 74:

`const top = stack[stack.length - 1];`